### PR TITLE
unti test for dassert_is_valid_timestamp

### PR DESCRIPTION
**Pull request by jayati1397**
Created this PR for Issue [#1094](https://github.com/kaizen-ai/kaizenflow/issues/1094).

**Changes Made:**
- Implemented unit tests for `dassert_is_valid_timestamp`

**Testing:**
Ran code through pylint and pytest.
- Pylint Errors: Class name: currently follows the convention specified [here](https://github.com/kaizen-ai/kaizenflow/blob/master/docs/coding/all.write_unit_tests.how_to_guide.md#naming-and-placement-conventions).

```
OUTPUT from the added lines in the file
helpers/test/test_hdatetime.py:781:0: C0115: Missing class docstring (missing-class-docstring)
helpers/test/test_hdatetime.py:781:0: C0103: Class name "Test_dassert_is_valid_timestamp" doesn't conform to PascalCase naming style (invalid-name)
```
- Pytest: No errors.
```
OUTPUT:
collected 3 items      
                                                                                                                                 
helpers/test/test_hdatetime.py::Test_dassert_is_valid_timestamp::test1 (0.00 s) PASSED [ 33%]
helpers/test/test_hdatetime.py::Test_dassert_is_valid_timestamp::test2 (0.00 s) PASSED [ 66%]
helpers/test/test_hdatetime.py::Test_dassert_is_valid_timestamp::test3 (0.00 s) PASSED [100%]

```

@samarth9008 please let me know if you have any further suggestions

### CODE REVIEW

File: helpers/test/test_hdatetime.py [hunk 1]
```diff
@@ -773,3 +773,44 @@ def test2(self) -> None:
             hdateti.dassert_str_is_date(date)
         actual = str(err.exception)
         self.check_string(actual)
+
+# #############################################################################
+# Test_dassert_is_valid_timestamp
+# #############################################################################
+
+class Test_dassert_is_valid_timestamp(hunitest.TestCase):
```
Comment (samarth9008) [inline helpers/test/test_hdatetime.py:781]: Can we add one more test of passing a timestamp as a string?

To test this LOC

```
hdbg.dassert_isinstance(timestamp, pd.Timestamp)
```
https://github.com/kaizen-ai/kaizenflow/blob/bc8bf878faa174a77c5d23fe7976de7307b1d717/helpers/hdatetime.py#L291C44-L291C56

**Review by samarth9008 — COMMENTED** [review]
Few nits from my side

File: helpers/test/test_hdatetime.py [hunk 2]
```diff
@@ -773,3 +773,44 @@ def test2(self) -> None:
...
+        """
+        timestamp = _PD_TS_UTC
+
+        # This should not raise any exceptions.
+        hdateti.dassert_is_valid_timestamp(timestamp)
+        
+
+    def test2(self) -> None:
+        """
+        Test dassert_timestamp_lt by checking a timestamp without a timezone
```
[the remark this answers is not in the document]
Comment (jayati1397) [reply helpers/test/test_hdatetime.py:795]: It is raising an exception

[the remark this answers is not in the document]
Comment (samarth9008) [reply helpers/test/test_hdatetime.py:795]: Yeah, I just want to change the comment. This is not something you need to check

File: helpers/test/test_hdatetime.py [hunk 3]
```diff
@@ -773,3 +773,57 @@ def test2(self) -> None:
...
+
+    def test1(self) -> None:
+        """
+        Test by checking a timestamp with a timezone.
+        """
+        timestamp = pd.Timestamp("2021-01-04 09:30:00-05:00", tz="America/New_York")
+        # This should not raise any exceptions.
+        hdateti.dassert_is_valid_timestamp(timestamp)
+        
+
```
Comment (samarth9008) [inline helpers/test/test_hdatetime.py:791]: Only one empty line between 2 function.

File: helpers/test/test_hdatetime.py [hunk 4]
```diff
@@ -773,3 +773,57 @@ def test2(self) -> None:
...
+        timestamp = pd.Timestamp("2021-01-04 09:30:00")
+        with self.assertRaises(AssertionError) as cm:
+            hdateti.dassert_is_valid_timestamp(timestamp)
+        act = str(cm.exception)
+        exp = """
+        * Failed assertion *
+        'None' is not 'None'
+        datetime_='2021-01-04 09:30:00' doesn't have timezone info
+        """
+        #check.
```
Comment (samarth9008) [inline helpers/test/test_hdatetime.py:805]: -> `# Check.`

Same everywhere.

File: helpers/test/test_hdatetime.py [hunk 5]
```diff
@@ -773,3 +773,57 @@ def test2(self) -> None:
...
+
+# #############################################################################
+# Test_dassert_is_valid_timestamp
+# #############################################################################
+
+class Test_dassert_is_valid_timestamp(hunitest.TestCase):
+
+    def test1(self) -> None:
+        """
+        Test by checking a timestamp with a timezone.
```
Comment (samarth9008) [inline helpers/test/test_hdatetime.py:785]: checking -> passing

File: helpers/test/test_hdatetime.py [hunk 6]
```diff
@@ -773,3 +773,57 @@ def test2(self) -> None:
...
+        * Failed assertion *
+        'None' is not 'None'
+        datetime_='2021-01-04 09:30:00' doesn't have timezone info
+        """
+        #check.
+        self.assert_equal(act, exp, fuzzy_match=True)
+
+    def test3(self) -> None:
+        """
+        Test by checking a timestamp as None.
```
Comment (samarth9008) [inline helpers/test/test_hdatetime.py:810]: checking -> passing

**Review by samarth9008 — COMMENTED** [review]
Final nits.

**Comment by samarth9008** [comment]
@jayati1397 Can you resolve conflicts on this PR and then its a merge?

**Comment by jayati1397** [comment]
> @jayati1397 Can you resolve conflicts on this PR and then its a merge?

Done!

**Pull request merged**